src/os/unix/wayland.rs

```diff
@@ -0,0 +1,1182 @@
+        event_q
+            .sync_roundtrip(&mut (), |_, _, _| unreachable!())
+            .map_err(|e| Error::WindowCreate(format!("Roundtrip failed: {:?}", e)))?;
+
+        //requires version 5 for the scroll events
+        let seat = global_man
+            .instantiate_exact::<WlSeat>(5)
+            .map_err(|e| Error::WindowCreate(format!("Failed retrieving the WlSeat: {:?}", e)))?;
+
+        //create the input devices already at this point
+        let input = WaylandInput::new(&seat);
+
+        //retrieve some types from globals
+        let comp = global_man
+            .instantiate_exact::<WlCompositor>(4)
+            .map_err(|e| {
+                Error::WindowCreate(format!("Failed retrieving the compositor: {:?}", e))
+            })?;
+        let shm = global_man.instantiate_exact::<WlShm>(1).map_err(|e| {
+            Error::WindowCreate(format!("Failed creating the shared memory: {:?}", e))
+        })?;
+
+        let surface = comp.create_surface();
+
+        //specify format
+        let format = if alpha {
+            Format::Argb8888
+        } else {
+            Format::Xrgb8888
+        };
+
+        //Retrive file to write to
+        let mut buf_pool = BufferPool::new(shm.clone(), format);
+        let (mut tmp_f, buffer) = buf_pool.get_buffer(size);
+
+        //Add a black canvas into the framebuffer
+        let frame: Vec<u32> = vec![0xFF000000; (size.0 * size.1) as usize];
+        let slice = unsafe {
+            std::slice::from_raw_parts(
+                frame[..].as_ptr() as *const u8,
+                frame.len() * std::mem::size_of::<u32>(),
+            )
+        };
+        tmp_f.write_all(&slice[..]).unwrap();
+        tmp_f.flush().unwrap();
+
+        let xdg_wm_base = global_man.instantiate_exact::<XdgWmBase>(1).map_err(|e| {
+            Error::WindowCreate(format!("Failed retrieving the XdgWmBase: {:?}", e))
+        })?;
+
+        //Ping Pong
+        xdg_wm_base.quick_assign(|xdg_wm_base, event, _| {
+            use wayland_protocols::xdg_shell::client::xdg_wm_base::Event;
+
+            if let Event::Ping { serial } = event {
+                xdg_wm_base.pong(serial);
+            }
+        });
+
+        let xdg_surface = xdg_wm_base.get_xdg_surface(&surface);
+        let _surface = surface.clone();
+        //Ping Pong
+        xdg_surface.quick_assign(move |xdg_surface, event, _| {
+            use wayland_protocols::xdg_shell::client::xdg_surface::Event;
+
+            if let Event::Configure { serial } = event {
+                xdg_surface.ack_configure(serial);
+                _surface.commit();
+            }
+        });
+
+        //Assigns the toplevel role and commit
+        let xdg_toplevel = xdg_surface.get_toplevel();
+        if decoration {
+            use wayland_protocols::unstable::xdg_decoration::v1::client::zxdg_decoration_manager_v1::ZxdgDecorationManagerV1;
+
+            if let Ok(deco_man) = global_man
```

Comment:
Again, more descriptive naming.

```suggestion
            if let Ok(decorations) = global_man
```